### PR TITLE
pass number of pca basis vectors used for ceres landmark and image co…

### PR DESCRIPTION
**Pull request by euklid**
…sts as parameters

Fixes two of the ToDo's in the code.

@patrikhuber please review.

### CODE REVIEW

File: examples/fit-model-ceres.cpp [hunk 1]
```diff
@@ -258,7 +258,7 @@ int main(int argc, char *argv[])
 	Problem camera_costfunction;
 	for (int i = 0; i < image_points.size(); ++i)
 	{
-		CostFunction* cost_function = new AutoDiffCostFunction<fitting::LandmarkCost, 2 /* num residuals */, 4 /* camera rotation (quaternion) */, num_cam_trans_intr_params /* camera translation & fov/frustum_scale */, 10 /* shape-coeffs */, 6 /* bs-coeffs */>(new fitting::LandmarkCost(morphable_model.get_shape_model(), blendshapes, image_points[i], vertex_indices[i], image.cols, image.rows, use_perspective));
+		CostFunction* cost_function = new AutoDiffCostFunction<fitting::LandmarkCost, 2 /* num residuals */, 4 /* camera rotation (quaternion) */, num_cam_trans_intr_params /* camera translation & fov/frustum_scale */, 10 /* shape-coeffs */, 6 /* bs-coeffs */>(new fitting::LandmarkCost(morphable_model.get_shape_model(), blendshapes, image_points[i], vertex_indices[i], 10, image.cols, image.rows, use_perspective));
```
Comment (patrikhuber) [inline examples/fit-model-ceres.cpp]: We're already passing the number of coefficients as a template argument. I'm wondering whether this duplication is necessary and we also need to pass them as function argument? (Same below for the shape & colour coefficients.)

**Comment by patrikhuber** [comment]
Hi!

Thank you for the PR! :-) Sorry for the late reply, I am quite busy until September.
It looks good to me! I've added a comment - if you could have a look at that, and then I'd say we can merge.

File: examples/fit-model-ceres.cpp (hunk 1 above)
Comment (euklid) [inline examples/fit-model-ceres.cpp]: I think the problem here is, that the template parameter `10` is for indicating the array size (==dimension) of input parameter of the shape coefficients when evaluating `operator()` of the `fitting::LandmarkCost` cost function that is put as an argument to the `AutoDiffCostFunction` and only the `AutoDiffCostFunction` knows the size only through the template argument and cannot give the size to the `operator()`. Thus, when trying to evaluate the model, we somehow need to know how many parameters are there and the only possibility would be at the constructor of `fitting::LandmarkCost` I think.

Comment (euklid) [inline examples/fit-model-ceres.cpp]: as a small refactoring however, we could introduce a variable in the main file so we don't have to hard code it.

**Comment by patrikhuber** [comment]
I've revamped the Ceres fitting in d101ca546118ae8736920915c6ac0dead448161f and it's now much neater and uses Eigen. There are now also `constexpr` variables to specify the number of shape, blendshape and colour coefficients at compile-time (https://github.com/patrikhuber/eos/blob/8a1c4ba768645314b342b2dc3636b5f8c76dd316/examples/fit-model-ceres.cpp#L197-L199). Sorry there wasn't any progress anymore on this PR back then.

**Pull request closed without merging**